src/types/submission.ts

```diff
@@ -35,37 +40,54 @@ export interface WebhookView {
   data: WebhookData
 }
 
+export interface ISubmissionSchema extends ISubmission, Document {
+  getWebhookView(): WebhookView | null
+}
+
 export interface IEmailSubmission extends ISubmission {
-  recipientEmails?: string[]
+  recipientEmails: string[]
   responseHash: string
   responseSalt: string
   hasBounced: boolean
+  encryptedContent: never
+  verifiedContent: never
+  version: never
+  attachmentMetadata: never
+  webhookResponses: never
+  getWebhookView(): WebhookView | null
 }
 
-export interface IEmailSubmissionSchema
-  extends IEmailSubmission,
-    ISubmissionSchema {}
+export type IEmailSubmissionSchema = IEmailSubmission & ISubmissionSchema
 
 export interface IEncryptedSubmission extends ISubmission {
+  recipientEmails: never
+  responseHash: never
+  responseSalt: never
+  hasBounced: never
   encryptedContent: string
   verifiedContent?: string
   version: number
   attachmentMetadata?: Map<string, string>
-  webhookResponses: IWebhookResponseSchema[]
+  webhookResponses: IWebhookResponse[]
+  getWebhookView(): WebhookView | null
 }
 
-export interface IWebhookResponseSchema extends Document {
+export type IEncryptedSubmissionSchema = IEncryptedSubmission &
+  ISubmissionSchema
+
+export interface IWebhookResponse {
   webhookUrl: string
   signature: string
-  errorMessage: string
-  response: {
-    status: number
-    statusText: string
-    headers: string
-    data: string
+  errorMessage?: string
+  response?: {
+    status?: number
+    statusText?: string
+    headers?: string
+    data?: string
```

Comment:
Our types are a subset of the official `AxiosResponse` type, which is why I chose to deliberately specify it as such

```
export interface AxiosResponse<T = any>  {
  data: T;
  status: number;
  statusText: string;
  headers: any;
  config: AxiosRequestConfig;
  request?: any;
}
```

Reply:
Can we use [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) or `Omit` instead?

Seems like we can use Omit:

```ts
response: Omit<AxiosResponse, 'config' | 'request'>
```

Reply:
I guess `headers` would be `any` instead of a stringified json? We do modify `data` to by stringifying it to.

Using `Omit<AxiosResponse<string>, 'config' | 'request'>` would mean being lose on `headers`? 
 @karrui @liangyuanruo

Reply:
I guess we can extend (and override headers) too, and data can be typed as `string` since `AxiosResponse`'s data is typed with a generic:

```ts
type SubmissionResponse = Omit<
  AxiosResponse<string>,
  'config' | 'request' | 'headers'
> & {
  headers: string
}
```

Reply:
great idea! done